src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWallet.scala

```diff
@@ -61,22 +58,22 @@ class ErgoWallet(historyReader: ErgoHistoryReader, settings: ErgoSettings)
     this
   }
 
-  override def rollback(to: VersionTag): Try[ErgoWallet] =
-    historyReader.heightOf(scorex.core.versionToId(to)) match {
-      case Some(height) =>
-        walletActor ! Rollback(to, height)
-        Success(this)
-      case None if to == ErgoState.genesisStateVersion =>
-        walletActor ! Rollback(to, ErgoHistory.GenesisHeight)
-        Success(this)
-      case None =>
-        Failure(new Exception(s"Height of a modifier with id $to not found"))
+  override def rollback(to: VersionTag): Try[ErgoWallet] = {
+    if (historyReader.heightOf(scorex.core.versionToId(to)).isDefined || to == ErgoState.genesisStateVersion) {
+      walletActor ! Rollback(to)
+      Success(this)
+    } else {
+      Failure(new Exception(s"Height of a modifier with id $to not found"))
```

Comment:
To understand the new code, I found useful to look at the old code, which suggests that the logical structure of the old code is more clear. Why this is changed?

Reply:
to remove boilerplate, two conditions had the same processing code, which is error-prone at least

Reply:
Hard to understand code is even more error prone.
Also this change doesn't look equivalent.

Reply:
It is equivalent, but ok, reverted